AwooChat.py

```diff
@@ -29,84 +29,76 @@ def titleWatchdog(msg):
         for i in AwooUtils.getChatAdmins(groupId):
             bot.sendMessage(i, "[ADM] A Group Name has been changed by " + offendingUsername + ". New name: " + offendingTitle)
 
+def parsePokemon(source):
+    parsedPokemon = {}
+    s = json.load(urllib2.urlopen(source))
+    pokemonList = s["pokemons"]
+
+    for i in pokemonList:
+        id = i['pokemon_id']
+        lat = i['latitude']
+        lon = i['longitude']
+        enc_id = i['encounter_id']
+        despawn = datetime.fromtimestamp(i['disappear_time'] / 1000).strftime('%H:%M:%S')
+
+        if (enc_id not in pokemonWatchdogSeen):
+            parsedPokemon[id] = parsedPokemon.get(id, [])
+            parsedPokemon[id].append({'lat': lat, 'lon': lon, 'despawn': despawn})
+            pokemonWatchdogSeen.append(enc_id)
+
 def pokemonWatchdog():
     global pokemonWatchdogSeen
-    p = prefs
 
-    now = int((datetime.now() - datetime(1970, 1, 1)).total_seconds() * 1000)
-    message = ""
+    now_time = int((datetime.now() - datetime(1970, 1, 1)).total_seconds() * 1000).time()
 
-    parsedPokemon = {}
+    pokemonBySource = {}
 
-    for chat_id in p.all():
-    
-        source = prefs.get(chat_id, 'pokemonDataSource', None)
-        if source is None:
-            continue
-            
-        source = source + "/raw_data"
-
-        s = json.load(urllib2.urlopen(source))
-        pokemonList = s["pokemons"]
-
-        for i in pokemonList:
-            id = i['pokemon_id']
-            lat = i['latitude']
-            lon = i['longitude']
-            enc_id = i['encounter_id']
-            despawn = datetime.fromtimestamp(i['disappear_time'] / 1000).strftime('%H:%M:%S')
-
-            if (enc_id not in pokemonWatchdogSeen):
-                parsedPokemon[id] = parsedPokemon.get(id, [])
-                parsedPokemon[id].append({'lat': lat, 'lon': lon, 'despawn': despawn})
-                pokemonWatchdogSeen.append(enc_id)
-                
-                
+    for chat_id in prefs.all():
         # Skip the program chats
         if chat_id == "__default" or chat_id == "__private" or chat_id == "__group":
             continue
-            
-        message = ""
-        alertList = p.get(chat_id, 'alertPokemonList')
 
-        if alertList is None:
+        source = prefs.get(chat_id, 'pokemonDataSource')
+        if not source:
+            continue
+
+        source += "/raw_data"
+
+        alertList = prefs.get(chat_id, 'alertPokemonList')
+        if not alertList:
```

Comment:
I moved this before parsing the source since there's no reason to parse the source if the `alertList` is empty.